src/halmos/calldata.py

```diff
@@ -79,33 +80,61 @@ def parse_tuple_type(var: str, items: list[dict]) -> Type:
 @dataclass(frozen=True)
 class EncodingResult:
     data: list[BitVecRef]
-    size: int
-    static: bool
+    size: int  # number of bytes
+    static: bool  # static vs dynamic type
+
+
+@dataclass(frozen=True)
+class DynamicParams:
```

Comment:
would we be better off with a `DynamicParam` dataclass?

```
@dataclass(frozen=True)
class DynamicParam:
    name: str
    size: int
    typ: Type
```

and `DynamicParams` becomes just a `list[DynamicParam]`?

Reply:
done in https://github.com/a16z/halmos/pull/360/commits/36ecb35bdd86e86093ceb08912c9087c8e3dd365